File: python/binary_search_iterative.py



def binary_search(l: list, x):
    low = 0
    high = len(l) - 1
    m = 0
    while low<=high:
        m = (low+high)//2
        if l[m]==x:
            return m
        elif l[m]>x:
            high = m-1
        else:
            low = m+1
    return -1

def main():
    l = [1, 5, 8, 14, 26, 100, 315, 643]
    print(binary_search(l, 100))

main()

File: cpp/binary_search_iterative.cpp
#include<iostream>
#include<vector>
using namespace std;

int binary_search(vector<int> vec, int x) {
	int low = 0;
	int high = vec.size()-1;
	while(low<=high) {
		int m = (high+low)/2;
		if(x<vec[m]) {
			high = m-1;
		}
		else if(x>vec[m]) {
			low = m+1;
		}
		else {
			return m;
		}
	}
	return -1;
}

int main() {
	vector<int> vec = {1, 13, 43, 94, 430, 1000, 15000};
	cout << binary_search(vec, 94) << endl;
	return 0;
}
